joint_limits_enforcement_plugins/test/test_ruckig_joint_limiter.cpp

```diff
@@ -12,33 +12,276 @@
+    desired_joint_states_.accelerations[0] = 5.0;
+
+    for (auto i = 0u; i < 4; ++i)
+    {
+      auto previous_vel_request = desired_joint_states_.velocities[0];
+      ASSERT_TRUE(joint_limiter_->enforce(current_joint_states_, desired_joint_states_, period));
+
+      ASSERT_LE(
+        desired_joint_states_.velocities[0],
+        previous_vel_request);  // vel adapted to reach end-stop should be decreasing
+      // NOTE: after the first cycle, vel is reduced and does not trigger stopping position limit
+      // hence no max deceleration anymore...
+      ASSERT_LT(
+        desired_joint_states_.positions[0],
+        5.0 + COMMON_THRESHOLD);  // should decelerate at each cycle and stay below limits
+      ASSERT_LE(desired_joint_states_.accelerations[0], 0.0);  // should decelerate
+
+      Integrate(period.seconds());
+
+      ASSERT_LT(current_joint_states_.positions[0], 5.0);  // below joint limit
+    }
+  }
+}
+
+TEST_F(RuckigJointLimiterTest, when_acceleration_exceeded_expect_acc_enforced)
+{
+  SetupNode("ruckig_joint_limiter");
+  Load();
+
+  if (joint_limiter_)
+  {
+    Init();
+    Configure();
+
+    rclcpp::Duration period(0, 5000000);  // 0.005 second
+
+    // desired acceleration exceeds
+    current_joint_states_.positions[0] = 0.0;
+    current_joint_states_.velocities[0] = 0.0;
+    desired_joint_states_.velocities[0] = 0.5;  // leads to acc > max acc
+    ASSERT_TRUE(joint_limiter_->enforce(current_joint_states_, desired_joint_states_, period));
+
+    // check if vel and acc limits applied
+    CHECK_STATE_SINGLE_JOINT(
+      desired_joint_states_, 0,
+      desired_joint_states_.positions[0],  // pos unchanged
+      // TODO(gwalck) why is the result divided by 2.0 ?
+      desired_joint_states_.accelerations[0] * period.seconds() /
+        2.0,                    // vel limited to max_acc * dt / 2.0
+      100.0 * period.seconds()  // acc is max_jerk * dt
+    );
+  }
+}
+
+TEST_F(RuckigJointLimiterTest, when_deceleration_exceeded_expect_dec_enforced)
+{
+  SetupNode("ruckig_joint_limiter");
+  Load();
+
+  if (joint_limiter_)
+  {
+    Init();
+    Configure();
+
+    rclcpp::Duration period(0, 5000000);  // 0.005 second
+
+    // desired deceleration exceeds
+    current_joint_states_.positions[0] = 0.0;
+    current_joint_states_.velocities[0] = 1.0;
+    desired_joint_states_.velocities[0] = 0.5;  // leads to acc > -max dec
+    ASSERT_TRUE(joint_limiter_->enforce(current_joint_states_, desired_joint_states_, period));
+
+    // check if vel and acc limits applied
+    CHECK_STATE_SINGLE_JOINT(
+      desired_joint_states_, 0,
+      desired_joint_states_.positions[0],  // pos unchanged
+      // TODO(gwalck) why is the result divided by 2.0 ?
+      desired_joint_states_.accelerations[0] * period.seconds() /
+        2.0,                     // vel limited to max_acc * dt / 2.0
+      -100.0 * period.seconds()  // acc is max_jerk * dt
+    );
+  }
+}
+/* ruckig does not use max_dec anyway
```

Comment:
can we set this? I think it is not possible, then OK; we can remove this and add a warning in the code if someone defines max_deceleration

Reply:
min_acceleration is part of Ruckig pro apparently (at least it is an std::optional and I had an error which told the variable does not exist) https://github.com/pantor/ruckig/blob/master/include/ruckig/input_parameter.hpp#L91C49-L91C65